sys-apps/yarn/yarn-1.15.2.ebuild

```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MY_P="${PN}-v${PV}"
+
+DESCRIPTION="Fast, reliable, and secure node dependency management"
+HOMEPAGE="https://yarnpkg.com"
+SRC_URI="https://github.com/yarnpkg/yarn/releases/download/v${PV}/${MY_P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="!dev-util/cmdtest
+	net-libs/nodejs"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${MY_P}"
+
+src_install() {
+	local install_dir="/usr/$(get_libdir)/node_modules/yarn" path shebang
+	insinto "${install_dir}"
+	doins -r .
+	dosym "../$(get_libdir)/node_modules/yarn/bin/yarn.js" "/usr/bin/yarn"
+
+	while read -r -d '' path; do
+		read -r shebang < ${path} || die
+		[[ "${shebang}" == \#\!* ]] || continue
+		chmod +x "${path}" || die #614094
```

Comment:
Could this be switched to fperms instead? How many files does this monster modify, can they be predicted and this cleaned?

Reply:
Nowadays, actually not too many:

```
$ equery files --tree yarn
 * Searching for yarn ...
 * Contents of sys-apps/yarn-1.15.2:
 /usr
   > /bin
      + yarn -> ../lib64/node_modules/yarn/bin/yarn.js
   > /lib64
      > /node_modules
         > /yarn
            + LICENSE
            + README.md
            > /bin
               + yarn
               + yarn.cmd
               + yarn.js
               + yarnpkg
               + yarnpkg.cmd
            > /lib
               + cli.js
               + v8-compile-cache.js
            + package.json
```

I modified it to use fperms anyway, keeping the loop. Could be done manually so it's more explicit in the ebuild.